internal/primitives/runtime/generic/header.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package generic
+
+import (
+	"io"
+
+	"github.com/ChainSafe/gossamer/internal/primitives/core/hash"
+	"github.com/ChainSafe/gossamer/internal/primitives/runtime"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// Header is a block header, and implements a compatible encoding to `sp_runtime::generic::Header`
+type Header[N runtime.Number, H runtime.Hash, Hasher runtime.Hasher[H]] struct {
```

Comment:
If this will be our new standard header type, the fields should be exported and then you can remove the Getters, since it will need to be scale encoded

Reply:
This needs to implement the `runtime.Header` which adopts the getter/setter pattern.

Reply:
Why is it needed for `runtime.Header` to implement this pattern if the types (and thus fields if we wanted)  will be exported anyway?

Reply:
😕 `generic.Header` implements `runtime.Header`? SCALE encoding is handled by implementing the `MarshalSCALE` and `UnmarshalSCALE` methods, so SCALE encoding is accounted for.  The fact that the number needs to encoded as `Compact<usize>` is implemented using the helper.

Reply:
Yeah i suppose I was wondering if there was a way to avoid using the helper struct for encoding/decoding, but I think this is a pretty small issue. Just wanted to make sure we had a good reason for needing the helper